lib/resty/wasmx/shm.lua

```diff
@@ -295,78 +320,186 @@ local function set_kv_value(zone, key, value, cas)
         error("value must be a string", 2)
     end
 
-    if type(cas) ~= "number" then
+    if cas == nil then
+        cas = 0
+
+    elseif type(cas) ~= "number" then
         error("cas must be a number", 2)
     end
 
     local shm = zone[WASM_SHM_KEY]
     local cname = ffi_new("ngx_str_t", { data = key, len = #key })
     local cvalue = ffi_new("ngx_str_t", { data = value, len = #value })
-    local written = ffi_new("uint32_t [1]")
+    local written = ffi_new("ngx_int_t[1]")
 
-    local rc = C.ngx_wa_ffi_shm_set_kv_value(shm, cname, cvalue, cas, written)
-    if rc == FFI_ERR then
-        return nil, "failed setting kv value, no memory"
+    local rc = C.ngx_wa_ffi_shm_kv_set(shm, cname, cvalue, cas, written)
+    if rc == FFI_ERROR then
+        return nil, "no memory"
+    end
+
+    if rc ~= FFI_OK then
+        return nil, "unknown error"
```

Comment:
The previous comment also applies here.
`ngx_wa_ffi_shm_kv_set` returns whatever `ngx_wa_shm_kv_set_locked` returns.
`ngx_wa_shm_kv_set_locked` returns either `NGX_ERROR` if there's not enough memory for the new value or `NGX_OK` if the new value is successfully set.